sourceaddrs/source_remote.go

```diff
@@ -15,6 +15,17 @@ type RemoteSource struct {
 	subPath string
 }
 
+type RemoteSourceInfo struct {
+	RemoteSource       RemoteSource
+	VersionDeprecation *Deprecation
```

Comment:
I think I'm unsatisfied that these new structs appear in this package but are not used by this package at all.

We should either change ParseRemoteSource/MakeRemoteSource to include deprecation data in order to simplify sourcebundle data structures... or we should move these types into sourcebundle and wrap RemoteSource as normal. I think I am in favor of the latter since deprecation data isn't a "source address"

Reply:
Ah when you put it like that I have to totally agree. Included 👍 https://github.com/hashicorp/go-slug/pull/59/commits/3fa65750f5638397e8c92aa9bc469e4d7d628c52